source/appModules/calculator.py

```diff
@@ -112,6 +128,16 @@ def event_UIA_notification(self, obj, nextHandler, displayString=None, activityI
 				and doNotAnnounceCalculatorResults
 			):
 				return
+			# #16573: when pasting, calculator sends a truncated UIA displayString
+			# (for example "Display is 1" when it should be "Display is 12.34").
+			# To fix this, we ignore the displayString sent via the UIA notification,
+			# and fetch the value directly from the UI element.
+			for child in resultElement.children:
+				if child.UIAAutomationId in ("CalculatorResults", "CalculatorAlwaysOnTopResults"):
+					speech.cancelSpeech()
+					speech.speakMessage(child.name)
```

Comment:
```suggestion
					ui.message(child.name)
```

Reply:
Also applies to braille handler call. The ui.message function will both speak and braille whatever you want NVDA to say, so you can replace the speech and braille call with ui.message function call.

Reply:
Great suggestion.  Done.